### DISCUSSION TITLE
Individual task result

### DISCUSSION THREAD

**Discussion by dpasupathi** [alitto/pond#48] [Q&A]
Hi,

I am trying to figure out a way to get results of all the tasks separately. Does this pond library supports it?  Currently i see that there is a error field which gets overridden. I am looking for something like result channel where all the task errors can be propagated back?

Thanks
Pasu

**Comment by alitto** [comment]
Hi!
We currently don't provide a way to collect all the "results" of tasks submitted to a pool because that can be implemented using channels in a quite simple and type-checked manner. Example:
```go
// Create a pool
pool := pond.New(10, 0, pond.MinWorkers(10))

// Create a buffered channel to collect the results
results := make(chan int, 1000)

// Submit 1000 tasks
for i := 0; i < 1000; i++ {
	n := i
	pool.Submit(func() {
	         // Send the result of this task to the channel
		results <- someComplexFn(n)
	})
}

// Stop the pool and wait for all submitted tasks to complete
pool.StopAndWait()

// Print task results
for result := range results {
    fmt.Println(result)
}

```

**Reply by dpasupathi** [reply]
Thanks @alitto
